src/main/java/uk/gov/hmcts/ccd/domain/service/common/CaseAccessGroupUtils.java

```diff
@@ -0,0 +1,236 @@
+                } catch (JsonProcessingException e) {
+                    throw new ValidationException(String.format(e.getMessage()));
+                }
+
+                caseDetails.getData().put(CASE_ACCESS_GROUPS, mergedNode);
+
+            } else {
+                caseDetails.getData().put(CASE_ACCESS_GROUPS, caseAccessGroupWithIdsNode);
+            }
+
+            LOG.debug("CASE ACCESS GROUPS : {} ", caseDetails.getData().get(CASE_ACCESS_GROUPS));
+        }
+
+        Map<String,JsonNode> caseDataClassificationWithCaseAccessGroup =
+            updateCaseDataClassificationWithCaseGroupAccess(caseDetails,caseDataService, caseTypeDefinition);
+        caseDetails.setDataClassification(caseDataClassificationWithCaseAccessGroup);
+    }
+
+    public Map<String, JsonNode> updateCaseDataClassificationWithCaseGroupAccess(
+        CaseDetails caseDetails,
+        CaseDataService caseDataService, CaseTypeDefinition caseTypeDefinition) {
+        Map<String, JsonNode> dataClassification = caseDetails.getDataClassification();
+        JsonNode caseAccessGroupJsonNode = caseDetails.getData().get(CASE_ACCESS_GROUPS);
+
+        Map<String,JsonNode> caseAccessGroupdata = null;
+        if (caseAccessGroupJsonNode != null) {
+            caseAccessGroupdata = Map.of(CASE_ACCESS_GROUPS, caseAccessGroupJsonNode);
+        }
+        Map<String, JsonNode> outputDataClassification = caseDetails.getDataClassification();
+        if (caseAccessGroupdata != null && !caseAccessGroupdata.isEmpty()) {
+            // generate just the CaseAccessGroups data classification from just CaseAccessGroups field data
+            Map<String, JsonNode> justCaseAccessGroupsDataClassification =
+                caseDataService.getDefaultSecurityClassifications(
+                caseTypeDefinition,
+                Map.of(CASE_ACCESS_GROUPS, caseAccessGroupdata.get(CASE_ACCESS_GROUPS)),
+                new HashMap<>()
+            );
+
+            // .. then clone current data classification and set the CaseAccessGroup classification
+            outputDataClassification = cloneOrNewJsonMap(dataClassification);
+            outputDataClassification.put(CASE_ACCESS_GROUPS,
+                justCaseAccessGroupsDataClassification.get(CASE_ACCESS_GROUPS));
+        }
+        LOG.debug("CaseAccessGroup data classification : {} ", outputDataClassification);
+        return outputDataClassification;
+    }
+
+    public JsonNode findOrganisationPolicyNodeForCaseRole(CaseDetails caseDetails, String caseRoleId) {
+        JsonNode caseRoleNode = caseDetails.getData().values().stream()
+            .filter(node -> node.get(ORG_POLICY_CASE_ASSIGNED_ROLE) != null
+                && node.get(ORG_POLICY_CASE_ASSIGNED_ROLE).asText().equalsIgnoreCase(caseRoleId))
+            .reduce((a, b) -> {
+                LOG.info("No Organisation found for CASE_ACCESS_GROUPS={} caseType={} version={} ORGANISATION={},"
```

Comment:
```suggestion
                LOG.debug("No Organisation found for CASE_ACCESS_GROUPS={} caseType={} version={} ORGANISATION={},"
```
Don't think we want this logging by default in Prod.